File: src/MediCore.API/MediCore.Domain/Entities/Pharmacy/PharmacySale.cs
using MediCore.Domain.Common;
using MediCore.Domain.Enums;

namespace MediCore.Domain.Entities.Pharmacy;

public class PharmacySale : BaseEntity
{
    public Guid TenantId { get; set; }
    public Guid BranchId { get; set; }
    public Guid WarehouseId { get; set; }
    public Guid? PatientId { get; set; }
    public Guid? PrescriptionId { get; set; }
    public Guid? InvoiceId { get; set; }
    public PharmacySaleType SaleType { get; set; }
    public PharmacySaleStatus Status { get; set; } = PharmacySaleStatus.Completed;
    public DateTime SaleDate { get; set; } = DateTime.UtcNow;
    public decimal TotalAmount { get; set; }
    public Guid SoldBy { get; set; }

    public List<PharmacySaleItem> Items { get; set; } = new();
}

public class PharmacySaleItem : BaseEntity
{
    public Guid PharmacySaleId { get; set; }
    public Guid ItemId { get; set; }
    public Guid? BatchId { get; set; }
    public Guid? PrescriptionItemId { get; set; }
    public decimal Quantity { get; set; }
    public decimal UnitPrice { get; set; }

    public PharmacySale PharmacySale { get; set; } = null!;
}

File: src/MediCore.Desktop/lib/features/pharmacy/domain/entities/pharmacy_sale.dart
class PharmacySaleItem {
  final String id;
  final String itemId;
  final String itemName;
  final double quantity;
  final double unitPrice;

  const PharmacySaleItem({
    required this.id,
    required this.itemId,
    required this.itemName,
    required this.quantity,
    required this.unitPrice,
  });

  double get lineTotal => quantity * unitPrice;
}

class PharmacySale {
  final String id;
  final String warehouseId;
  final String? patientId;
  final String? prescriptionId;
  final String? invoiceId;
  final String saleType;
  final String status;
  final DateTime saleDate;
  final double totalAmount;
  final List<PharmacySaleItem> items;

  const PharmacySale({
    required this.id,
    required this.warehouseId,
    this.patientId,
    this.prescriptionId,
    this.invoiceId,
    required this.saleType,
    required this.status,
    required this.saleDate,
    required this.totalAmount,
    required this.items,
  });
}

/// A line the pharmacist is building up before submitting the sale -
/// local UI state, not sent to the backend until CreateSale.
class PharmacyCartLine {
  final String itemId;
  final String itemName;
  final double unitPrice;
  final double quantity;
  final String? prescriptionItemId;

  const PharmacyCartLine({
    required this.itemId,
    required this.itemName,
    required this.unitPrice,
    required this.quantity,
    this.prescriptionItemId,
  });

  double get lineTotal => quantity * unitPrice;
}
